### PR TITLE
To preserve data in JSON Files

### PR DESCRIPTION
**Pull request by r-ahmed2022**
In this PR, We completed the following:-
- On startup, the app load all data from files. , and handle the case when one or more files are missing.
- On exit, the app saves all the data (books, people, and rentals) into JSON files. 
- Tested the app

### CODE REVIEW

**Comment by MahmoudBakr23** [comment]
# Hi team😃, 

As much as I appreciate your hard work on this project, I'm afraid that I can't perform a review on this pull request.

## **Invalid Submission**

- Please make sure to resolve the conflicts of the recently merged branch and run the Linters checks again to the last commit. 🙏 

_Your Code Review Request will be marked as invalid in your Dashboard, so please submit a new one once you are ready._ 🙏

Cheers and Happy coding!👏👏👏

Feel free to mention @MahmoudBakr23 to leave any questions or comments in the PR thread if something is not 100% clear.

------
_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._

<hr>

_Invalid Code Review Request does not count into the code reviews limit._

File: app.rb [hunk 1]
```diff
@@ -4,17 +4,18 @@
...
-require_relative 'options'
-
-
+require './write_read'
 class App
   def initialize
     @books = []
     @people = []
     @rentals = []
+
```
Comment (BregornOriginal) [inline app.rb:13]: - Kindly fix the error when a user tries to run the application and there's no JSON files already created

For example, if in the program there's nothing stored or already created, the program automatically should create the necessary file when you create a user or books or rental, but shouldn't give you an error while trying to run the application without JSON files created

**Review by BregornOriginal — CHANGES_REQUESTED** [review]
## Hi @Team 👏🏼 👏🏼 👏🏼 

## Good job so far!

## There are some issues that you still need to work on to go to the next project but you are almost there!



## Highlights

✅ Linters are passing
✅ Good commit history
✅ Data has been stored on json files 🥇 

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Comment by Kashif-Saleem-Ghuman** [comment]
@BregornOriginal  Hi, Thank you for reviewing our PR 👍 . However, I don't understand the issue that you are trying to point out. I have made a video where I deleted all the data files and then when I run the app it creates the required files when we create a person, book, or rental without giving any errors.
https://www.loom.com/share/fb71636e8d114a2d90d8db2c65d11956

File: app.rb (hunk 1 above)
Comment (r-ahmed2022) [reply app.rb:13]: @BregornOriginal ,  the partner has posted the video of the same, deleting all the files and starting from the scratch, Kindly elaborate further..

**Review by Lameck1 — COMMENTED** [review]
Hello Team 👋,

_This is an additional review as a result of the second opinion request._

From my investigation, the app works as expected. I started with all the JSON files deleted, the app ran and persisted data. I exited the app and ran it again, it was still functioning as expected. I can't seem to reproduce the bug that @BregornOriginal is talking about.

@BregornOriginal, when you encounter any error when testing an app, please remember to provide a screenshot of it in your code review.

As this was the only reason that your project was not approved it will be approved as soon as possible.
PLEASE DON'T ask for another CODE REVIEW, we'll manually approve this from our side and you'll be able to submit this task ASAP.

Keep rocking 🚀!

**Comment by Kashif-Saleem-Ghuman** [comment]
@Lameck1  Thank you very much 😄

**Review by Munsa1 — APPROVED** [review]
Hi @r-ahmed2022 @Kashif-Saleem-Ghuman , 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

## Highlights
Resolved conflicts :heavy_check_mark: 
Correct GitHub Flow :heavy_check_mark: 
 

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Comment by Kashif-Saleem-Ghuman** [comment]
@Munsa1  Thank you very much for the approval 👍

**Comment by BregornOriginal** [comment]
> Hello Team 👋,
> 
> _This is an additional review as a result of the second opinion request._
> 
> From my investigation, the app works as expected. I started with all the JSON files deleted, the app ran and persisted data. I exited the app and ran it again, it was still functioning as expected. I can't seem to reproduce the bug that @BregornOriginal is talking about.
> 
> @BregornOriginal, when you encounter any error when testing an app, please remember to provide a screenshot of it in your code review.
> 
> As this was the only reason that your project was not approved it will be approved as soon as possible. PLEASE DON'T ask for another CODE REVIEW, we'll manually approve this from our side and you'll be able to submit this task ASAP.
> 
> Keep rocking 🚀!

Hi @Lameck1 thanks for helping the student, I couldn't answer before because time zone. So, yes the app works correctly. 
It's would be something wrong on my side yesterday. But I just update VsCode today and works perfectly. Excuse me for the misunderstanding. 

***Great job! @Kashif-Saleem-Ghuman*** 👏🏼 👏🏼 👏🏼

**Pull request merged**